src/server/cmd/job-shim/main.go

```diff
@@ -54,6 +85,27 @@ func do(appEnvObj interface{}) error {
 				return err
 			}
 
+			// Start sending ContinuePod to PPS to signal that we are alive
+			go func() {
+				tick := time.Tick(10 * time.Second)
+				for {
+					<-tick
+					res, err := ppsClient.ContinuePod(
+						context.Background(),
+						&ppsserver.ContinuePodRequest{
+							ChunkID: response.ChunkID,
+							PodName: appEnv.PodName,
+						},
+					)
+					if err != nil {
+						lion.Errorf("error from ContinuePod: %s", err.Error())
+					}
+					if res != nil && res.Exit {
+						os.Exit(0)
```

Comment:
This seems like a potential bug, `os.Exit` will cause the program to exit immediately bypassing all deferred function calls. There's 1 later in this function that unmounts the mountpoint which won't run if we hit this line. I think that could potentially cause the pod to not clean up correctly, I've seen leaked fuse mounts cause some problems like that.